cgroupstats.py

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) 2017-present, Outlyer
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+
+"""Cgroupstats module
+
+This module exists to expose the cgroupstats api to python
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+# @lint-avoid-python-3-compatibility-imports
+# from __future__ import unicode_literals
+
+import struct
+import gnlpy.netlink as netlink
+
+
+class LoadStats(object):
+    __fields__ = [
+        'nr_sleeping', 'nr_running', 'nr_stopped',
+        'nr_uninterruptible', 'nr_iowait'
+    ]
+
+    def __init__(self, **kwargs):
+        self.__dict__.update(kwargs)
+
+    def __repr__(self):
+        arr = ['%s=%s' % (f, repr(self.__dict__[f])) for f in self.__fields__]
+        return 'LoadStats(%s)' % ', '.join(arr)
+
+    @staticmethod
+    def unpack(val):
+        fmt = 'QQQQQ'  # as per __fields__ above, and they're uint64 each
+        val = val[4:]  # strip attribute length, first 4 bytes
+        attrs = dict(zip(LoadStats.__fields__, struct.unpack(fmt, val)))
+        return LoadStats(**attrs)
+
+
+CgroupstatsCmdAttrList = netlink.create_attr_list_type(
+    'CgroupstatsAttrList',
+    ('CGROUPSTATS_CMD_ATTR_FD', netlink.U32Type),
+)
+
+
+CgroupstatsMessage = netlink.create_genl_message_type(
+    'CgroupstatsMessage', 'TASKSTATS',
+    ('NULL', netlink.IgnoreType),  # Reserved
+    ('GET', netlink.IgnoreType),
+    ('NEW', netlink.IgnoreType),
```

Comment:
IIRC, the kernel will emit a `NEW` event back, and this what is going to be used to trigger the decapsulation. So I think, what we would want here is:
```
     ('NEW', CgroupstatsType),
```

where `CgroupstatsType` is similar to `TaskstatsType`, which is made of `netlink.U64Type` type. With each fields corresponding to their equivalent in `LoadStats`, e.g `NR_SLEEPING`, `NR_RUNNING`....

`LoadStats` will become unnecessary as there is no need for complex types.